macaroonbakery/discharge.py

```diff
@@ -129,7 +135,15 @@ def discharge(ctx, id, caveat, key, checker, locator):
     cond, arg = checkers.parse_caveat(cav_info.condition)
 
     if cond == checkers.COND_NEED_DECLARED:
-        cav_info.condition = arg.encode('utf-8')
+        cav_info = ThirdPartyCaveatInfo(
```

Comment:
cav_info = cav_info._replace(condition=arg.encode('utf-8'))

?

Reply:
a namedtuple is immutable unless you use protected method like this one, I prefer to create a new instance.

Reply:
the _replace method does return a new instance - that's the point. Using _replace is both cleaner and less error-prone IMHO.